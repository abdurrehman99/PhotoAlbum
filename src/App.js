import 'react-native-gesture-handler';
import React from 'react';
import store from './redux/store';
import {Provider} from 'react-redux';
import AppNavigation from './navigation';

const App = () => {
  return (
    <Provider store={store}>
      <AppNavigation />
    </Provider>
  );
};

export default App;
