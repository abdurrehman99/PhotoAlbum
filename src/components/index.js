import AlbumCard from './AlbumCard';
import Header from './Header';
import ImageThumbnail from './ImageThumbnail';
import SwipeButton from './SwipeButton';
import Backdrop from './Backdrop';

export {AlbumCard, Header, ImageThumbnail, SwipeButton, Backdrop};
