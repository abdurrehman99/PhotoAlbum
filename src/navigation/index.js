import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import StackNavigation from './StackNavigation';

const Navigation = (props) => {
	return (
		<NavigationContainer>
			<StackNavigation />
		</NavigationContainer>
	);
};

export default Navigation;
