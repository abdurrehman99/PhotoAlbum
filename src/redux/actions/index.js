import {getAlbumData, setAlbum, setAlbumsList} from './AlbumActions';
import {toggleFilter} from './FilterActions';

export {getAlbumData, setAlbumsList, setAlbum, toggleFilter};
