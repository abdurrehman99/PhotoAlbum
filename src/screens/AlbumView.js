import React from 'react';
import {ScrollView} from 'react-native';

const AlbumView = () => {
	return <ScrollView></ScrollView>;
};

export default AlbumView;
