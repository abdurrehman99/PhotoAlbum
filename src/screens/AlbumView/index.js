import React, {useState} from 'react';
import {ScrollView, View, FlatList, Text} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {connect} from 'react-redux';
import {Header, ImageThumbnail, SwipeButton} from '../../components';
import {styles} from './styles';

const AlbumView = () => {
  const [activeImage, setActiveImage] = useState(1);
  let imageURL = 'https://picsum.photos/709';
  return (
    <View style={styles.container}>
      <Header text={'Photo Album App'} />
      <View style={styles.imageContainer}>
        <ImageThumbnail fullWidth uri={imageURL} />
        <SwipeButton style={styles.rightIcon} right />
        <SwipeButton style={styles.leftIcon} left />
      </View>
      <FlatList
        data={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]}
        numColumns={3}
        scrollEnabled
        renderItem={({item, index}) => (
          <TouchableOpacity
            style={
              index === activeImage ? styles.activeThumbnail : styles.thumbnail
            }
            onPress={() => setActiveImage(index)}>
            <ImageThumbnail thumbnail uri={'https://picsum.photos/706'} />
          </TouchableOpacity>
        )}
        keyExtractor={(item, index) => index.toString()}
      />
    </View>
  );
};

const mapStateToProps = (state) => {
  return {};
};

const mapDispatchToProps = (dispatch) => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(AlbumView);
