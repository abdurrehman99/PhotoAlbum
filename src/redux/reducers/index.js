import {combineReducers} from 'redux';
import AlbumsReducer from './AlbumsReducer';
import FilterReducer from './FilterReducer';
import LoaderReducer from './LoaderReducer';

export default combineReducers({
  albumsData: AlbumsReducer,
  filter: FilterReducer,
  loading: LoaderReducer,
});
