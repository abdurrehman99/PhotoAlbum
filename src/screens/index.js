import AlbumView from './AlbumView';
import Home from './Home';

export {AlbumView, Home};
