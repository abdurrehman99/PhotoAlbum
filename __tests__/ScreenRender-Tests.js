import React from 'react';
import {Provider} from 'react-redux';
import {create, act} from 'react-test-renderer';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../src/redux/reducers';
import {Home} from '../src/screens';

let reduxInitialState = {
  albumsData: {
    albumsList: [],
    singleAlbum: {},
  },
  filter: {
    showFilter: false,
  },
  loading: false,
};

const store = createStore(
  rootReducer,
  reduxInitialState,
  applyMiddleware(thunk),
);

let props = {
  navigation: {navigate: jest.fn()},
};

const HomeScreen = create(
  <Provider store={store}>
    <Home {...props} />
  </Provider>,
);

const ViewAlbumScreen = create(
  <Provider store={store}>
    <Home {...props} />
  </Provider>,
);

jest.runAllTimers();

test('Render Home Screen', () => {
  expect(HomeScreen).toMatchSnapshot();
});

test('Render View Album Screen', () => {
  expect(ViewAlbumScreen).toMatchSnapshot();
});

// test('View Album button press', () => {
//   const ViewAlbumButton = HomeScreen.root.findByProps({
//     testID: 'ViewAlbumButton',
//   });

//   ViewAlbumButton.onPress();

//   expect(props.navigation.navigate).toBeCalledWith('AlbumView');
// });
