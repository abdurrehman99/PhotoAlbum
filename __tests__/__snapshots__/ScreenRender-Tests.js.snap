// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home Screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 5,
          "marginHorizontal": 75,
          "marginVertical": 133.4,
          "padding": 35,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginBottom": 15,
          }
        }
      >
        Filter Albums
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "purple",
                },
              ]
            }
          >
            Clear Filter
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "justifyContent": "center",
        "padding": 15,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFF",
          "borderRadius": 25,
          "padding": 10,
          "width": 262.5,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 16,
          }
        }
      >
        Home
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 15,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingLeft": 5,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Render View Album Screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 5,
          "marginHorizontal": 75,
          "marginVertical": 133.4,
          "padding": 35,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginBottom": 15,
          }
        }
      >
        Filter Albums
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "purple",
                },
              ]
            }
          >
            Clear Filter
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "justifyContent": "center",
        "padding": 15,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFF",
          "borderRadius": 25,
          "padding": 10,
          "width": 262.5,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 16,
          }
        }
      >
        Home
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 15,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingLeft": 5,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
